src/isomorphic/classic/element/ReactElement.js

```diff
@@ -123,8 +126,10 @@ ReactElement.createElement = function(type, config, children) {
   var source = null;
 
   if (config != null) {
-    ref = config.ref === undefined ? null : config.ref;
-    key = config.key === undefined ? null : '' + config.key;
+    ref = !config.hasOwnProperty('ref') ||
+      Object.getOwnPropertyDescriptor(config, 'ref').get ? null : config.ref;
```

Comment:
This should definitely not run in prod. This is a pretty weird condition to begin with though. I don't have anything better off the top of my head but it is weird that all getters here will behave differently.

Also, this is a behavior change if config.ref is set to `undefined` (`null` before, now `undefined`).